### DISCUSSION TITLE
Separate json and jsonc/json5 formatting

### DISCUSSION THREAD

**Discussion by unional** [biomejs/biome#1620] [Ideas]
```json
{
	"files": {
		"include": ["**/*.json"]
	},
	"json": {
		"parser": {
			"allowComments": true,
			"allowTrailingCommas": true
		}
	}
}
```

This setting will apply to all json files including files that does not support comments or trailing commas.

Those are features in jsonc/json5 and there should be a way so keep the two (three) kinds separated.

Each of them needs to support their own `include` and `ignore` field.

**Comment by ematipico** [comment]
> This setting will apply to all json files including files that does not support comments or trailing commas.

I think there's a bit of confusion/misunderstanding.

JSON doesn't support comments or trailing commas. Although, some tools such as VSCode are tolerant and allow comments and trailing comma in some of **its files**. It parses those JSON files as they were JSONC files 

In an attempt to meet the community halfway, we added those settings.

In your case, if you want to be tolerant about trailing commas and comments **only for a specific set of files**, you need to use `overrides`:  https://biomejs.dev/reference/configuration/#overrides

**Reply by unional** [reply]
I see. Thx.
So if I want to make the config accurate for all json files, I should turn those off at the base, and override them for all those offenders. 🤣

**Reply by ematipico** [reply]
Yes exactly!
